constraints.md

```diff
@@ -148,45 +133,155 @@ CREATE TABLE users (
+
+## FOREIGN KEY
+
+> **Note:**
+>
+> TiDB has limited support for foreign key constraints.
+
+TiDB supports creating `FOREIGN KEY` constraints in DDL commands.
+
+For example:
 
 ```sql
-DROP TABLE IF EXISTS users;
 CREATE TABLE users (
  id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
- username VARCHAR(60) NOT NULL,
- UNIQUE KEY (username)
+ doc JSON
 );
-INSERT INTO users (username) VALUES ('dave'), ('sarah'), ('bill');
+CREATE TABLE orders (
+ id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
+ user_id INT NOT NULL,
+ doc JSON,
+ FOREIGN KEY fk_user_id (user_id) REFERENCES users(id)
+);
+```
 
-mysql> SET tidb_constraint_check_in_place = TRUE;
-Query OK, 0 rows affected (0.00 sec)
+{{< copyable "sql" >}}
 
-mysql> START TRANSACTION;
-Query OK, 0 rows affected (0.00 sec)
+```sql
+SELECT table_name, column_name, constraint_name, referenced_table_name, referenced_column_name
+FROM information_schema.key_column_usage WHERE table_name IN ('users', 'orders');
+```
 
-mysql> INSERT INTO users (username) VALUES ('jane'), ('chris'), ('bill');
-ERROR 1062 (23000): Duplicate entry 'bill' for key 'username'
+```
++------------+-------------+-----------------+-----------------------+------------------------+
+| table_name | column_name | constraint_name | referenced_table_name | referenced_column_name |
++------------+-------------+-----------------+-----------------------+------------------------+
+| users      | id          | PRIMARY         | NULL                  | NULL                   |
+| orders     | id          | PRIMARY         | NULL                  | NULL                   |
+| orders     | user_id     | fk_user_id      | users                 | id                     |
++------------+-------------+-----------------+-----------------------+------------------------+
+3 rows in set (0.00 sec)
+```
 
-..
+TiDB also supports the syntax to `DROP FOREIGN KEY` and `ADD FOREIGN KEY` via the `ALTER TABLE` command.
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE orders DROP FOREIGN KEY fk_user_id;
+ALTER TABLE orders ADD FOREIGN KEY fk_user_id (user_id) REFERENCES users(id);
 ```
 
-* The first `INSERT` statement causes a duplicate key error. This results in additional network communication, and will likely decrease insert throughput.
+### Notes
+
+* TiDB supports foreign keys to avoid errors caused by this syntax when you migrate data from other databases to TiDB.
+
+    However, TiDB does not perform constraint checking on foreign keys in DML statements. For example, even if there is no record with id=123 in the users table, the following transactions can be submitted successfully.
+
+    ```sql
+    START TRANSACTION;
+    INSERT INTO orders (user_id, doc) VALUES (123, NULL);
+    COMMIT;
+    ```
+
+* TiDB does not display foreign key information in the result of executing the `SHOW CREATE TABLE` statement.
```

Comment:
This may be fixed in https://github.com/pingcap/tidb/pull/18360 before your PR merges. It will still be relevant to TiDB 4.0 and earlier, but not master.

Reply:
I think this PR can be merged within today. We can update the related content later when #18360 is merged.😄